### PR TITLE
Improve readability of link text on dark backgrounds #12606

### PR DESCRIPTION
**Pull request by katebee**
## What does this change?

Add different link styling to links within a `.tonal--tone-media` div. 
This is a fix for issue #12606 
## What is the value of this and can you measure success?

Increased readability of text while keeping links apparent. Style is consistent (and based on) the links in [gallery captions](http://www.theguardian.com/artanddesign/gallery/2016/apr/23/the-20-photographs-of-the-week); these also appear on a dark background.

Examples of the **targeted links** can be found [here](http://www.theguardian.com/books/picture/2016/apr/19/re-illustrating-the-jungle-book-monkey-kidnapping-by-sarah-mcintyre-and-philip-reeve) and [here](http://www.theguardian.com/books/picture/2016/apr/22/re-illustrating-the-jungle-book-jungle-lessons-by-marta-altes)

This change does not affect captions unless they have an ancestor with the class `.tonal--tone-media`; an **unaffected link** can be found on [this page](http://www.theguardian.com/world/2016/jan/04/sunni-allies-join-saudi-arabia-in-ceasing-diplomatic-ties-with-iran)
## Screenshots

**BEFORE**
before 1
**AFTER**
after 1
**BEFORE**
before 2
**AFTER**
after 2
## Request for comment

@paperboyo

### CODE REVIEW

**Comment by katebee** [comment]
Using both classes to add the link styling is necessary; the **Reuse this content** button is nested within the `.tonal--tone-media` div. Without further selecting for `.caption--img`, the button acquires text-decoration...

These appear to be the two classes with sufficient specificity to make the fix, however, discussion welcome! 😸

**Comment by paperboyo** [comment]
Thanks a  lot, @katebee! I can't read the code, so can't give :+1:, but can definitely give :fireworks:!

File: static/src/stylesheets/base/_links.scss [hunk 1]
```diff
@@ -12,4 +12,9 @@ a,
         color: $news-accent;
         text-decoration: none;
     }
+    // Improve readability of link text in figcaptions that are on dark backgrounds (issue #12606)
+    .tonal--tone-media .caption--img & {
+        color: $neutral-7;
+        border-bottom: 1px solid $neutral-2;
```
Comment (sndrs) [inline static/src/stylesheets/base/_links.scss]: do we need border bottom? `text-decoration: underline` should do it?

**Comment by katebee** [comment]
Using `border-bottom: 1px solid $neutral-2;` gives these links subtle hover and focus events; the underline becomes slightly more vibrant. This mimics the behaviour of the bulleted link above the images and of the links in [the galleries](http://www.theguardian.com/money/gallery/2016/apr/27/outrageously-expensive-homes-in-pictures)

If `text-decoration: underline` is used on line 18, the link appearance does not change on hover.  

Style was copied from [_gallery.scss](https://github.com/guardian/frontend/blob/113bb0101e70e1805969c6384821241be56cbac6/static/src/stylesheets/module/content/_gallery.scss) for `.gallery2__caption a`

What style and link effects are preferred?

File: static/src/stylesheets/base/_links.scss [hunk 2]
```diff
@@ -12,4 +12,9 @@ a,
         color: $news-accent;
         text-decoration: none;
     }
+    // Improve readability of link text in figcaptions that are on dark backgrounds (issue #12606)
+    .tonal--tone-media .caption--img & {
```
Comment (gtrufitt) [inline static/src/stylesheets/base/_links.scss]: I think tone related styles should be in the tone scss file and not base? https://github.com/guardian/frontend/blob/master/static/src/stylesheets/module/content/tones/_tone-media.scss

File: static/src/stylesheets/module/content/tones/_tone-media.scss [hunk 3]
```diff
@@ -48,6 +48,13 @@
         border-color: $neutral-2;
     }
 
+    .caption--img {
+        a {
```
Comment (gtrufitt) [inline static/src/stylesheets/module/content/tones/_tone-media.scss]: Don't need to nest this ie: `.caption--img a`

**Comment by gtrufitt** [comment]
Other than minor nesting comment, all good! 👍

File: static/src/stylesheets/module/content/tones/_tone-media.scss [hunk 4]
```diff
@@ -48,6 +48,13 @@
...
+            color: $neutral-7;
+            border-bottom: 1px solid $neutral-2;
+        }
+    }
+
```
Comment (OliverJAsh) [inline static/src/stylesheets/module/content/tones/_tone-media.scss]: Are there other links on the page whose styles we could share/re-use here? Links in the standfirst are the same, I think?

**Comment by NataliaLKB** [comment]
@katebee before merging just take a look with how it works with @zeftilldeath 's changes in https://github.com/guardian/frontend/pull/12846.

**Comment by johnduffell** [comment]
28 days old! 😱 this has a +1 so any reason not to ship?

**Comment by NataliaLKB** [comment]
@johnduffell: @katebee is on holiday for another week and a bit

**Comment by johnduffell** [comment]
closing for now, please reopen when it's ready to go

**Comment by paperboyo** [comment]
The fact that I'm on holidays (not for much longer) should not stop anyone! :laughing:

**Pull request closed without merging**